src/Kmd.Logic.Gateway.Automation/Publish.cs

```diff
@@ -102,22 +102,84 @@ private async Task CreateOrUpdateProducts(Guid subscriptionId, Guid providerId,
             foreach (var productValidationResult in productValidationResults)
```

Comment:
I feel this class is growing beyond what it should. Probably this is the right time to refactor the responsibility into smaller ones that take care of products, apis and so on.. Not a review comment as such. But just a thought!!

Reply:
You are definitely right. We can make a task for that.

Reply:
Created.